### DISCUSSION TITLE
Action failed to cancel

### DISCUSSION THREAD

**Discussion by zaherg** [community/community#25570] [Actions]
<p>Not sure what is the issue here but I can’t seems able to cancel this workflow <a href="https://github.com/linuxjuggler/laravel-mysql-test/commit/1332f47d9ea60955c91ef2f7eb34a42d0a56548c/checks?check_suite_id=247611483" rel="nofollow noopener">https://github.com/linuxjuggler/laravel-mysql-test/commit/1332f47d9ea60955c91ef2f7eb34a42d0a56548c/checks?check_suite_id=247611483</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4544e17a86fdcfdf3539bc8bb051d16c5c3d81f5.png" data-download-href="/uploads/short-url/9SMwlXIB0AKFNEabkY0SQk4HPjD.png?dl=1" title="Screen Shot 2019-10-02 at 10.26.19 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/27624/181091851-640ac458-455e-4f5e-9063-3e341e6d6d85.png" alt="Screen Shot 2019-10-02 at 10.26.19 AM.png" data-base62-sha1="9SMwlXIB0AKFNEabkY0SQk4HPjD" width="690" height="195" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4544e17a86fdcfdf3539bc8bb051d16c5c3d81f5_2_690x195.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/4544e17a86fdcfdf3539bc8bb051d16c5c3d81f5_2_1035x292.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/4544e17a86fdcfdf3539bc8bb051d16c5c3d81f5_2_1380x390.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4544e17a86fdcfdf3539bc8bb051d16c5c3d81f5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-10-02 at 10.26.19 AM.png</span><span class="informations">1892×537 66.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by riker09** [comment]
<p>I have an action that is running for two days now. I am not able to cancel it. Clicking on “Cancel run” only yields the “Failed to cancel check suite.” error as shown in the first post (attached image).</p>
<p>So, same here.</p>

**Reply by P-r-e-m-i-u-m** [reply]
It looks like this issue was caused by the repository or workflow being deleted, which is why the cancel option wasn’t working properly. Once everything was removed, the problem resolved itself. So in this case, it seems like there’s nothing more to worry about and things are back to normal now.

**Comment by zaherg** [comment]
<p>Any idea how to fix this issue? It has been three days now.</p>

**Comment by zaherg** [comment]
<p>hey <a href="/t5/user/viewprofilepage/user-id/38776">@chrispat</a>  and sorry for bothering you, but is there anyone who can look at this issue? and I’ve read a few people complains about the same issue.</p>
<p>This is my 6th day and the job didn’t stop yet.</p>
<p>Thanks again.</p>

**Comment by riker09** [comment]
<p>I believe they (the guys at GitHub) are aware of this issue. I don’t see any harm in having a workflow in the running state for indefinite time. It most likely was killed by some garbage collection process a long time ago.</p>
<p>I would like to be able to immediately cancel any running workflow, however. I haven’t been able to do this with any running workflow.</p>

**Comment by zaherg** [comment]
<p>Right now, since they are not counting the running time, this is okay, but someone should look at it :D.</p>
<p>I didn’t face any problem with any actions I used, and an error in the yml file causes this one here.</p>

**Comment by gregoil** [comment]
<p>Same here</p>
<p><a href="https://github.com/gregoil/rotest/commit/54bd1dd400e22ad03a031daa7bdd3494b64bb5fd/checks?check_suite_id=289395292" rel="nofollow noopener">https://github.com/gregoil/rotest/commit/54bd1dd400e22ad03a031daa7bdd3494b64bb5fd/checks?check_suite_id=289395292</a></p>

**Comment by zaherg** [comment, marked as answer]
<p>I think they have deleted everything so now everything is okay.</p>

**Reply by MohamedElhawy** [reply]
It's far from "okay" with me. How can I rest when there is a workflow still running for days 😟, your ticket is from 2019 and now it's 2023 ?! why aren't they fixing it ?

**Reply by hellohejinyu** [reply]
I also face it now

**Reply by edy-rodrigues** [reply]
This problem happened to me 1 hour ago. I believe there was a problem with the Github Actions system

**Reply by hellohejinyu** [reply]
<img src="https://user-images.githubusercontent.com/8766034/222606632-995609cf-3654-4167-bbc9-88d8c1a0d518.jpeg" width="200" />

😿

**Reply by trent-abc** [reply]
Here's a scary thought, tried to cancel a workflow that attempts to delete things, like a runaway freight train it plows through to completion ignoring any cancellation attempts

**Reply by jserpapinto** [reply]
2024 and we have 2 workflows stuck after an outage on Github Actions..

**Reply by razortanmayshah** [reply]
Same occurred with us also. Github support team, please resolve this ASAP

**Reply by lucaspdroz** [reply]
It's 2026... and still same Issue... please, use a engeneer no an dumb AI to solve this ASAP!

**Reply by sayaliM0412** [reply]
Reporting as of today May 15,2026 !! Same issue. 

<img width="602" height="214" alt="Screenshot 2026-05-15 at 12 30 04 PM" src="https://github.com/user-attachments/assets/e0d53785-e5ea-4f48-8bb9-c046bbcd0ba4" />

**Comment by softprops** [comment]
<p>I’ve run into this issue as well in a private repo except that an action that runs in 2s has been running for 18minutes with all workflow runs triggered by pushes after queued. </p>
<p>I believe the issue might be related to trying to rerun a job that failed but needs a passing job to trigger. The rerun logic may not be accounting for the passing job and stuck in a waiting state</p>

**Comment by bfloch** [comment]
<p>Just openend this:</p>
<p><a href="https://github.community/t/action-stuck-for-4-days-with-local-runner-windows/16859">https://github.community/t5/GitHub-Actions/Action-stuck-for-4-days-with-Local-Runner-Windows/m-p/37895#M3119</a></p>
<p>In my case it is a single local runner - so this basically blocks any further work.</p>

**Comment by softprops** [comment]
<p>I’ve tried a few tactics and none of them worked. I’ve tried renaming my workflow as well as completely deleting and recreating my repository (which is okay it was brand new) but still the repository workflow seemed indefinitely queued</p>

**Comment by zaherg** [comment]
<p><span class="mention">@softpropsI</span> advise you to contact the support since now your running actions counted towards your quota.</p>

**Comment by softprops** [comment]
<p>A came back the next day and my workflow run queue back to normal. I can’t say what changed so know don’t know if this will happen again</p>

**Comment by carloscortonc** [comment]
Hi! Is this still a problem on Github? After a failed action start (https://github.com/carloscortonc/cli-er/actions/runs/2921222337/attempts/1), and a re-run (https://github.com/carloscortonc/cli-er/actions/runs/2921222337/attempts/2, I only recall executing "re-run" once) both 2nd and 3rd executions hang at "queued" state, and trying to cancel latest run results in "Failed to cancel workflow."

**Comment by runxel** [comment]
<del>Still having this issue....urgh.</del>
<ins>EDIT: Looks like it will be force terminated 5 minutes after you hit that cancel button, even if it was not successful in the first place.</ins>

**Reply by jlaustill** [reply]
Not happening for me :(

**Reply by ettaboyle** [reply]
Hi @runxel ,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Comment by aseem-heg** [comment]
![Screenshot from 2023-03-02 17-37-02](https://user-images.githubusercontent.com/86262107/222588975-aaad29df-e758-45f7-a916-b4e33f6ab5f9.png)
When I try to cancel a workflow - I get this screen but the workflow never gets cancelled.

**Reply by ettaboyle** [reply]
Hi @aseem-heg ,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Comment by webcubed** [comment]
My actions are not working either. Trying to deploy my website through github pages, been stuck for a while now.

**Reply by ettaboyle** [reply]
Hi @oofish8888,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Comment by RTS-Galaxies** [comment]
Same here, failing to cancel or run jobs

"This page is taking too long to load."

Still not working =(

**Reply by fireai-developer** [reply]
they're working on it. 🤞 
![image](https://user-images.githubusercontent.com/102819437/222598568-c215a3d8-e489-4240-9a0d-bd34db74d815.png)
(from https://www.githubstatus.com/)

**Reply by scottrudy** [reply]
Actions show as being resolved, but I still have an action stuck that I can't cancel

![image](https://user-images.githubusercontent.com/627993/222634557-f0f43882-1729-4046-99f2-a82aa4a3d5ff.png)

**Reply by frankiejarrett** [reply]
Same here 😞 

Still cannot cancel a workflow that has been stuck "Queued" for 7 hours.

I get the red `Failed to cancel workflow.` error.

**Reply by shannonjarcher** [reply]
Still experiencing the same issue. I can't rerun the damned thing while its queued.

**Reply by ettaboyle** [reply]
Hi @shannonjarcher @fjarrett @scottrudy @fireai-developer @RTS-Galaxies 

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Reply by fireai-developer** [reply]
Been using it all morning without any issues. Only a handful jobs from yesterday are 'stale' and cannot be cancelled as per comments above. Thanks for fixing the main issue right away. 🌞

**Reply by ettaboyle** [reply]
Thanks for the updates!!

**Comment by silviosampaio** [comment]
Same here! 

![Screen Shot 2023-03-03 at 09 54 12](https://user-images.githubusercontent.com/78871242/222725313-c7d3da0b-7604-43df-9716-d97750291c61.png)

**Reply by ettaboyle** [reply]
Hi @silviosampaio,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Reply by silviosampaio** [reply]
@ettaboyle It's working for me now! Thanks!

**Reply by ettaboyle** [reply]
Nice!! Thanks for letting us know

**Comment by ghost** [comment]
Lots and lots of folks seem to be experiencing this. Github status page now says problem is resolved, but jobs are still stuck in queues with no opportunity to cancel. A word from github at this point would be good.

**Reply by ettaboyle** [reply]
Hi @jrte,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Comment by jlaustill** [comment]
We are also experiencing this, and it is blocking us from deploying. The timing could not be worse either; please update with a workaround or ETA.

**Reply by ettaboyle** [reply]
Hi @jlaustill ,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Comment by BMayhew** [comment]
Experiencing the same thing, we have 5 jobs that are in queue and I am unable to cancel any of the jobs. One of these is a deploy job of an older git version of code. If these get deleted I'm all good, but if something automatically kicks off we will have an older version of code deployed to production 😨

**Reply by ettaboyle** [reply]
Hi @BMayhew,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Reply by scottrudy** [reply]
@ettaboyle The status page doesn't address what will happen with these actions that are currently queued and not able to be canceled. As @BMayhew stated, if these actions end up running they are going to overwrite more recent deployments, which is likely not want anyone wants. We're just looking for an answer on what will happen with them.

**Reply by Akash1134** [reply]
@scottrudy This issue has been escalated to the engineering team and is currently being reviewed. We'll get back to you as soon as we have more details to share. Do note since the overall job has been completed, it should not affect your billing. If you notice any discrepancies do let us know and we will investigate further. In addition to that, if the workflow logs do not provide enough detail to diagnose why a workflow, job, or step is not working as expected, you can [enable additional debug logging](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging).

**Comment by imran-ag** [comment]
Same experience

**Reply by ettaboyle** [reply]
Hi @imran-ag,

Thanks for sharing and we appreciate your patience. You can check our [GitHub Status Page](https://www.githubstatus.com/) to see for real-time and historical data on system performance + subscribe to issues for updates.

**Current Status:** [Investigating reports of degraded performance](https://www.githubstatus.com/incidents/sr7zptjd0b07)

**Reply by imran-ag** [reply]
@ettaboyle Github says the problem is resolved however we still have a pending action in our workflow. It doesnt allow us to cancel it either. Do you have any solution for this?

**Reply by ettaboyle** [reply]
You can [cancel the whole workflow](https://docs.github.com/en/actions/managing-workflow-runs/canceling-a-workflow) and I'm looking into this to see if you have the ability to cancel a pending action. 
cc'ing @Akash1134 who is the Community manager for Actions. I was tagging in to help with the status announcement last week. Akash - I tagged you in the issue. 🕵🏻‍♂️

**Reply by Akash1134** [reply]
Hi @imran-ag , Sorry that you're still having trouble running your workflow. As @ettaboyle mentioned, please re-run the action in order to cancel the workflow. We have received other reports where any pending workflows stuck during the incident have not resumed or cannot be cancelled. You can also use the tools in GitHub Actions to [monitor](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/about-monitoring-and-troubleshooting#monitoring-your-workflows) and [troubleshoot](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/about-monitoring-and-troubleshooting#troubleshooting-your-workflows) your workflows. While troubleshooting your workflows, if the workflow logs do not provide enough detail to diagnose why a workflow, job, or step is not working as expected, you can [enable additional debug logging](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging).

**Comment by kalinkochnev** [comment]
I'm having this issue

**Reply by ScottBurger** [reply]
One thing that just worked for me now was restarting the DNS check for the custom domain in the GitHub Pages settings menu. My pages-build-deployment was stuck, and I noticed that the custom domain check was stuck on "DNS Check in Progress". I set the domain to some fake website, hit save, let the DNS check fail, then reset it back to my real domain and everything unstuck itself.

Not sure if that applies here, but I thought I'd add some more data to the solution if possible.

**Comment by MeltedYeti** [comment]
This is still an issue I've had an action running for a couple hours now and all attempts to cancel have returned an error.

**Comment by itaibh** [comment]
I made an accidental release because the cancel didn't work.
At first I thought it'll stop right away.
Then I thought it'll stop after the current task.
Then I thought it'll stop after the current job.
But it never stopped and just continued until the unwanted release.

**Comment by prasannareddys** [comment]
it is still same issue. `Triggered via pull request 7 months ago`

**Reply by Blitheness** [reply]
There is now an API endpoint to force cancellation of a workflow: https://github.blog/changelog/2023-09-21-github-actions-force-cancel-workflows/

**Reply by joshua-temple** [reply]
> There is now an API endpoint to force cancellation of a workflow: https://github.blog/changelog/2023-09-21-github-actions-force-cancel-workflows/

I have a workflow that has been stuck for three weeks, and the above endpoint unfortunately yields no difference.

```
  "message": "Failed to cancel workflow run",
  "documentation_url": "https://docs.github.com/rest/actions/workflow-runs#force-cancel-a-workflow-run",
  "status": "500"
```

**Comment by ihtkas** [comment]
We are also facing the same issue. Workflows older than 3 weeks are still in Queued state. We tried to cancel using UI, cancel as well as force-cancel APIs. All of them return the "Failed to cancel workflow run" error. Our GitHub checks are now broken completely.

**Comment by karmicdude** [comment]
The same problem here. Doesn't react to workflow cancel via UI, doesn't react to `force-cancel` via API with status code 500 and `"Failed to cancel workflow run"` message. Deletion of workflow run also doesn't work with same error.

**Comment by chaseisabelle** [comment]
Same here. 3 days in "queued", trying to cancel yields "failed to cancel" error 😞

**Comment by johnlomat** [comment]
Same issue here

**Comment by Eric2XU** [comment]
same here... and I am fairly sure its messing up a action that otherwise has run for years every 5 mins without an issue.

**Comment by joshua-temple** [comment]
![Screenshot 2024-07-01 at 8 36 56 PM](https://github.com/community/community/assets/115025696/0b00c3ca-a240-4d18-94bb-4cd322189bb3)

Add another one to the list.. (it normally takes 20minutes to complete)

**Comment by Eric2XU** [comment]
FWIW, I opened a support case and they were able to force close mine. They said it was due to the run being a "check suite run". No idea what that is but you can make use of the [Update a check run](https://docs.github.com/en/enterprise-cloud@latest/rest/checks/runs?apiVersion=2022-11-28#update-a-check-run) API endpoint to close one out if found

**Comment by gianantoniopini** [comment]
I'm also facing the same issue. I tried making a `cancel` and `force-cancel` request to the REST API, but both requests fail with status 409 and message "Cannot cancel a workflow re-run that has not yet queued.". Which is confusing because my action has status "Queued" in the UI.

**Comment by meditator3** [comment]
same issue. what's going on? the github status page doesn't show any errors. this is a 4 weeks old workflow that can't be canceled.

**Comment by hdavila146** [comment]
here I have the same problem just now, the job is already complete and still in status process I try to cancel by GUI,  by API none worked and the force cancel too nothing worked. is a little frustrating because this consume the credits I think... anyone got any other solution...

**Comment by osamajandali** [comment]
I have the same error and it is preventing me from deploy, I need to deploy urgently

**Reply by Ibrahom1** [reply]
it should be good now as i was doing the same. Seems like the GitHub actions and services were down for some time. You can check https://www.githubstatus.com/ they update it if there are any issues

**Reply by osamajandali** [reply]
Oh yes it is working, thanks so much for letting me know

**Reply by meditator3** [reply]
mine's still turning and turning....
i feel bad for the runner

**Comment by Justina-03** [comment]
Same here... It doesn't work even though github status says "All Systems Operational".
![스크린샷 2024-10-13 115226](https://github.com/user-attachments/assets/7e32f918-560a-48fa-a722-9b4c16c967b6)

**Comment by tahaJemmali** [comment]
<img width="1583" alt="Screenshot 2024-10-30 at 10 58 38" src="https://github.com/user-attachments/assets/440a7832-1e8c-451b-a705-9aecae231662">
Same issue, I tried to cancel still running

**Comment by razvanabagiu89** [comment]
Same issue for me in 2025

**Comment by liquidcms** [comment]
Not sure same issue documented here, but perhaps:

I have an action which was never queued. It simply states Queued (days ago). Trying to cancel the run and i get "Failed to cancel workflow".

I am able to run other actions but any time i run this action it also gets stuck trying to queue. I suspect the first one being hung is hanging any other run of same action.

I tried the GET posted in link above. It returns 404.

**Comment by AFMiziara** [comment]
Since 2019 !!!!!!!!! We have enterprise plan on GitHub and no support either. Looks like the team is not worried at all in fixing issues, the fact that this goes on for a long time and there is no solution still is unbelievable.

**Comment by prmntr** [comment]
How is this still happening?? I tried to rerun a workflow which should take less than 30 seconds and has been queued for half an hour; even going through the API to try and cancel it yields no results.

**Comment by Ryangr0** [comment]
These are probably all different issues, just surfacing at the same place in the UI. For me, it's that the job is waiting to get picked up. It's been 2 hours. It's blocking my pipeline. I triggered another run with

```
concurrency:
  group: push-${{ github.branch }}
  cancel-in-progress: true
```

,

But that doesn't cancel it either. It did cancel other jobs of its kind that were queued up.

I can't progress doing what I was doing, because the job I need to run isn't running right now, even though I fixed the original bug I'm facing. (My github runners being out of date and refusing to pick up jobs). I can't run the job again with ubuntu-latest instead because the previous job, trying to run on my own runners, `arc-runner-set`, is eternally waiting to get picked up by runners that will never spawn. I'm unable to force this job to cancel for some reason that isn't being surfaced to me, the user. I hope this will resolve itself, it looks like it's completely out of my control for now?

**Comment by marcinkubica** [comment]
Problem is still there. I have product development and release affected. Please dig into the root cause.

 I've used `report off-topic` to admins option in hope Microsoft will pay attention and fix the issue, which has been "resolved" in 2019 by "they have deleted everything". Did this as I was linked by the Github Support site when creating the ticket as a place providing resolution to my issue.
 
 
![image](https://github.com/user-attachments/assets/f848b954-f04c-4f65-b574-5edb159177b9)
![image](https://github.com/user-attachments/assets/15869d5a-64f0-4d75-b7a6-d5606183e95e)

Thanks and best regards

**Comment by Hanchuxiao** [comment]
wtf the problem is still the same in nearly 2026?! It has been over 5 years !!!

**Comment by jsschrstrcks1** [comment]
its 2026 and I have a number of these running constantly in my repository.

**Comment by P-r-e-m-i-u-m** [comment]
It looks like this issue was caused by the repository or workflow being deleted, which is why the cancel option wasn’t working properly. Once everything was removed, the problem resolved itself. So in this case, it seems like there’s nothing more to worry about and things are back to normal now.

**Comment by marcinkubica** [comment]
for anyone who enters here ;)

```sh
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  /repos/${org}/${repo}/actions/runs/${runs}/force-cancel
  ```
  
  if above doesn't work you have to contact github support as the last resort is they delete a failed entry at the backend.

**Comment by UsaRandom** [comment]
same issue here, blocking job queue

https://github.com/pepecoinppc/pepecoin/actions/runs/17021279274

**Comment by DuckDuckStudio** [comment]
A similar situation, but the queued job is shown as failed on the left side of the page and in the notifications:
![image](https://github.com/user-attachments/assets/d51cd383-ae7b-4d28-a0c7-94f17a6ce59d)
![image](https://github.com/user-attachments/assets/52de75bd-6a01-4337-ad64-023fa9694567)
![image](https://github.com/user-attachments/assets/22c9b923-bcf1-40a6-9910-1b0431d78a99)
